### PR TITLE
Translations ( with context ) Simple CRUD UI

### PR DESCRIPTION
**Pull request by e-lobo**
1. Add new page Renovation Translation Tool.. ( accessible by http://0.0.0.0:8000/app/renovation_translation_tool ) ✅ 
2. Read existing translations ✅ 
3. Create New translation ✅ 
4. Edit existing translations ✅ 
5. Delete a translation ✅ 
6. Docs ✅

### CODE REVIEW

**Comment by fahimalizain** [comment]
@assemmarwan @e-lobo should we sort this based on the language code rather than the name of the language ? It's hard to find Arabic lol 😄 

image

File: renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.js [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+            }
+        });
+    }
+
+    show_add_new_translation() {
+        this.translation_btn_added = true
+        /**
+         We need the following to add a translation with context
+         key:doctype:name:fieldname
+         key:doctype:name
```
Comment (fahimalizain) [inline renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.js:166]: @e-lobo please update to include `*`

File: renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.js [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+                this.page.btn_primary.trigger("click")
+            });
+    }
+
+    load_translations() {
+        // load translations
+        if (this.selected_doctype && this.selected_language) {
+            frappe.call({
+                module: "renovation_core.renovation_core",
+                page: "renovation_translation_tool",
```
Comment (fahimalizain) [inline renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.js:457]: 🔥 🔥

File: renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    if doctype and docname and not docfield:
+        context_filters_conditions += f" or context LIKE '{doctype}:{docname}%'"
+
+    where_conditions += f" and ( {context_filters_conditions} )"
+
+    sql = """
+    SELECT  `tabTranslation`.name,
+            SUBSTRING_INDEX(context, ':', 1)  as 'document_type',
+            CASE LENGTH(context) - LENGTH(REPLACE(context, ':', ''))
+            WHEN 2 THEN SUBSTR(context, LOCATE(':', context, 8) + 1,
```
Comment (fahimalizain) [inline renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.py:97]: 🔥 🔥 🔥 
Documentation for this magic would be nice for anyone reading this in the future 😄

**Review by fahimalizain — CHANGES_REQUESTED** [review]

File: renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.py (hunk 3 above)
Comment (e-lobo) [reply renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.py]: > uld we sort this based on t

refactored 👍

**Comment by e-lobo** [comment]
> @assemmarwan @e-lobo should we sort this based on the language code rather than the name of the language ? It's hard to find Arabic lol 😄
> 
> image

refactored 👍

File: renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.py (hunk 3 above)
Comment (e-lobo) [reply renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.py]: done 👍

File: renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.js (hunk 1 above)
Comment (e-lobo) [reply renovation_core/renovation_core/page/renovation_translation_tool/renovation_translation_tool.js:178]: done 👍

**Review by fahimalizain — APPROVED** [review]
All Good 👍🏼 
Fix the XSS issue if you feel like 😄

**Review by assemmarwan — APPROVED** [review]

**Pull request merged**